### DISCUSSION TITLE
Trying to use urlMask but getting empty string

### DISCUSSION THREAD

**Discussion by rmuller-vu** [mia-platform/community#549] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.102.8

### Description

I am trying to use urlMask to inject an id to be used with the event `add-filter`. When I go to use the id it is an empty string. I want the 2nd filter to be able to use the id from the urlMask. I have tried both `/version-approval/:id` and `version-approval/:id` for the urlMask

url: `mfe-application/version-approval/6685548221585cd2c280907f`

```json
{
  "tag": "bk-button",
  "properties": {
    "urlMask": "version-approval/:id",
    "content": "All Needed Versions",
    "action": {
      "type": "event",
      "config": {
        "events": [
          {
            "label": "add-filter",
            "payload": {
              "operator": "equal",
              "property": "approvedOn",
              "value": null,
              "name": "approvedOnFilter"
            }
          },
          {
            "label": "add-filter",
            "payload": {
              "operator": "equal",
              "property": "projectId",
              "value": "{{pathnameParams.params.id}}",
              "name": "projectIdFilter"
            }
          }
        ]
      }
    }
  }
}
```

### Actual Outcome

The value of the `projectIdFilter` filter added is ""

### Expected Outcome

The value of the `projectIdFilter` filter added should be `6685548221585cd2c280907f`

**Comment by LiliumCandidum** [comment, marked as answer]
Hi @rmuller-vu,

Your `urlMask` should match the entire path, so if your path is `mfe-application/version-approval/6685548221585cd2c280907f`, your `urlMask` should be `/mfe-application/version-approval/:id`

**Reply by rmuller-vu** [reply]
This worked for me. Thank you!
